packages/gatsby-source-filesystem/src/create-lazy-file-node.js

```diff
@@ -0,0 +1,143 @@
+"use strict"
+
+const path = require(`path`)
+
+const mime = require(`mime`)
+const {
+  getRemoteFileExtension,
+  getRemoteFileName,
+  createFilePath,
+} = require(`./utils`)
+
+const { createContentDigest, slash } = require(`gatsby-core-utils`)
+const CACHE_DIR = `.cache`
+const FS_PLUGIN_DIR = `gatsby-source-filesystem`
+
+function createFileNode(pathToFile, createNodeId, pluginOptions = {}) {
+  const slashed = slash(pathToFile)
+  const parsedSlashed = path.parse(slashed)
+  const slashedFile = Object.assign({}, parsedSlashed, {
+    absolutePath: slashed,
+    // Useful for limiting graphql query with certain parent directory
+    relativeDirectory: slash(
+      path.relative(pluginOptions.path || process.cwd(), parsedSlashed.dir)
+    ),
+  })
+  let internal
```

Comment:
Haha. Well, I actually copied this section from create-file-node, because I wanted to be sure the parsing rules are the same. I'll refactor it into a helper function eventually.